torch/__init__.py

```diff
@@ -2292,11 +2292,15 @@ def compiled_with_cxx11_abi() -> builtins.bool:
 class _TorchCompileInductorWrapper:
     compiler_name = "inductor"
 
-    def __init__(self, mode, options, dynamic):
+    def __init__(self, mode, options, dynamic, package):
         from torch._inductor.compiler_bisector import CompilerBisector
 
         self.config: dict[str, _Any] = {}
         self.dynamic = dynamic
+        self.package = package
+        if self.package:
+            self.apply_options({"cpp_wrapper": True})
+            self.apply_options({"aot_inductor.package": True})
```

Comment:
IMO the initial version of this should use the Python wrapper code.  I think this will be required (at least in the short term) to support:
1) Training
2) Dynamic shapes
3) Graph breaks

I think the use case of single-graph, static-shape, inference is already well covered by export+AOTI and doesn't really need a new API.

Reply:
I think if we refactor the package code with AOTAutogradCacheEntry properly, we can combine these designs nicely into a single framework:
- Caching creates an AOTAutogradCacheEntry with Python wrapper cache entries 
- Precompile creates an AOTAutogradCacheEntry with either CPP wrapper or python wrapper cache entries, + a set of serialized dynamo guards. 

The nice thing about this is, if we do it right, we also get regular caching support for cpp wrapper, which we actually don't have today.

Reply:
@jansel I'm all for training as well. Just disagree with this

> inference is already well covered by export+AOTI and doesn't really need a new API.

inference is not well covered by export+AOTI since the UX is still broken for compiling partial graphs (wrote a post about this https://fb.workplace.com/groups/257735836456307/permalink/802058152024070/)

I'd like to still take inference into account, and as @jamesjwu mentioned, we can leverage caching to support both cases.

Reply:
I was talking about "single-graph, static-shape, inference" (all three at the same time) being well covered.